gf-webapp-admin/src/component/rooms/room/services/Services.tsx

```diff
@@ -1,35 +1,27 @@
 import {useState} from "react";
-import {Room} from "src/model/Room/RoomDeprecated";
-import {saveRoom} from "src/service/RoomService";
-import {Button} from "gf-ui-lib/components/Button/Button";
 import {changeRoomServices} from "./roomServices";
 import {ApartmentServices} from "src/model/Room/ApartmentServices";
 import {enumToArray} from "src/utils/enumToArray";
+import {useRoomContext} from "src/component/rooms/room/roomContext";
 import styles from "src/component/rooms/room/services/Services.module.scss";
 
-interface ServicesProps {
-  room:Room;
-}
-export const Services = (props: ServicesProps) => {
+export const Services = () => {
 
-  const [services, setServices] = useState(props.room.services);
-  const [isEditFieldDisabled, setIsEditFieldDisabled] = useState(true);
-
-  const saveHandler = () => {
-    setIsEditFieldDisabled(true);
-    props.room.services = services;
-    saveRoom(props.room);
-  };
-
-  const fieldEditHandler = () => {
-    setIsEditFieldDisabled(false);
-  };
+  const {room, setRoom} = useRoomContext();
+  const [services, setServices] = useState(room.services);
 
   const onChangeRoomServices = (e:React.ChangeEvent<HTMLInputElement>) => {
     const selectedService = e.target.value as unknown as ApartmentServices;
```

Comment:
we mustn't use as unknown as ...

Reply:
Unknown has been removed. `e.target.value` is not a constant. It can be any string, but with `as` we define this string as Apartments. So it is ok in this case.

Reply:
Maybe this is better way - https://stackoverflow.com/questions/56034705/how-to-add-enum-type-for-value-of-htmlinputelement/56037344#56037344 (mean to use user defined type guards)